### ISSUE TITLE
Crash error on android

### ISSUE THREAD

**Issue by pierroo** [gre/react-native-view-shot#507]
### bug report 

I get this error when I try to "viewShotRefs?.current.capture().then..."
![image](https://github.com/gre/react-native-view-shot/assets/17619412/6eed934e-61fd-4028-83b9-8b3b7c1e15cc)

obviously there may be something wrong in my view that I have yet to find, but why does it crash?
it should simply go into the }).catch(e => { etc

#### Version & Platform

v3.7.0
RN 0.69.12

**Platform:** Android

**Comment by songxiaoliang** [comment]
Same question

**Comment by songxiaoliang** [comment]
@pierroo solve it?

**Comment by pierroo** [comment]
@songxiaoliang yup, not a sexy fix but I simply wrapped the capture in a 300ms timeout and it fixed it.
no clue why, let me know if it worked for you too?

**Comment by songxiaoliang** [comment]
@pierroo oh, no

**Comment by songxiaoliang** [comment]
@pierroo I've been practicing the implementation you've given me, and I haven't found any problems so far, thanks!

**Comment by gre** [comment]
A lot has changed since v3.7.0 and RN 0.69. Please try `npm install react-native-view-shot@alpha` with a recent RN version. Stable v5 coming soon.

**Comment by gre** [comment]
Closing as outdated. You're on `react-native-view-shot@3.7.0` / RN 0.69 — both several major versions behind. v5.0.0 requires RN >=0.76 and ships proper exception propagation on Android: errors thrown inside the capture path now reject the promise (with the underlying Throwable as the cause) instead of crashing the executor (#628 / #630 / #631).

Please upgrade to RN >=0.76 and `react-native-view-shot@5.0.0+`, then re-test. If you still see a hard crash instead of a `.catch()` rejection, please open a fresh issue with: RN version, view-shot version, the offending view tree, and a logcat stack trace.

Release notes: https://github.com/gre/react-native-view-shot/releases/tag/v5.0.0
